### DISCUSSION TITLE
Pulsar Plus 3 Phase: Total minimum current

### DISCUSSION THREAD

**Discussion by andre-richter** [lbbrhzn/ocpp#276] [Q&A] [answered]
Hi,

I hope it is okay to ask here: I cannot find any information about the **total** minimum configurable current that you can set with 3 phase Pulsar Plus wallboxes.
Is it 6A _per phase_ or 6A for all three phases together? This would make quite a difference if you plan to only charge with photovoltaic excess.

Maybe somebody has checked that on his Pulsar already? :)

Thanks!

**Comment by GrantKeymer** [comment]
Hi Andre-Richter,

I have a Wallbox Pulsar Plus and there are no settings in the app or My Wallbox portal relating to 3-phase.  The circuit board has all 3 phases present even though I cannot use them because my EV doesn't support 3-phase charging.  Our local Wallbox distributor told me that every Pulsar Plus will support 3-phase charging, simply by changing the cable connecting to the EV.

In addition, I found this...

**Applicable to 1-phase and 3-phase connections**

"This smart Wallbox charging station is suitable for charging your car with a maximum of 3x32A (22 kW), but you can set this in values ​​between 6A and 32A so that this charging station is perfectly applicable to any network connection and can supply different charging capacities. This allows you to charge any electric car!"

To me, it looks as though the max. current setting ranges from 6A _per phase_ to 32A _per phase_.

Do you have a Pulsar Plus installed yet?

Cheers,
Grant.

**Reply by andre-richter** [reply]
Hi @GrantKeymer,

thanks for your feedback. I don't have a Pulsar Plus yet. I am currently checking out options.

My prime use case is that I want to heavily use solar excess power, and as little grid power as possible. Meaning that throughout the day, I will have something between 0 and ~5000 W excess available.

Should the `6A` setting of the pulsar plus apply to all three phases, I understand this as being able to set a minimum charging power of `6A * 230V * 3 = 4140 W`. However, this much excess will be very rarely available from solar, meaning I have to get the remaining part from the grid most of the time.

In a single phase setting, where the minimum charging power would be `6A * 230V = 1380 W`, it would be way easier to dynamically control the excess usage, because on sunny days, 1000 W or more is available even in the morning hours.

I have been told some wallboxes just switch off one or two phases if you want to go down to `6A` current overall. Unsure if the Pulsar would do that 🤔 

Best,
Andre

**Comment by GrantKeymer** [comment, marked as answer]
Hi @andre-richter,

Thanks for explaining further about your use case.  It's very similar to mine, actually.

We have 7560Wp of solar panels installed and during summer (now in New Zealand) excess solar power is plentiful.  It's no problem to fully charge our 40kWh Leaf on a fine day.  However, if it's partly cloudy, I find myself constantly watching the sky and adjusting the charge current accordingly.  My sole reason for joining this forum is to develop some sort of automation process which monitors excess solar power and adjusts the Pulsar Plus charging current accordingly.

I don't understand why the EV manufacturers decided to adopt a minimum charge current of 6 Amps.  When our Leaf is approaching 100% charge, and doing cell-balancing, the charge power fluctuates between 0.5kW and 1.2kW in cycles over a period of 30 to 60 minutes.  If it's OK to have a charging current of 2 or 3 Amps during cell-balancing, then why not all the time?

It doesn't make any sense to me, but that's what we are stuck with :(

In your case, maybe you should consider only connecting a single phase into the Pulsar Plus.  That would allow you to go as low as 1380W charging power, which it seems is the most important factor for you.  The trade-off is that you would lose access to the full 22kW charging capability.

Can your EV actually handle 22kW of charging power though?
My Leaf is restricted to 3.3kW and I find that is plenty, given the amount of driving we typically do in a day.

I am certain that the Pulsar Plus doesn't have the ability to switch off one or two phases.  I have studied the documents for the Pulsar Plus NA which can use split-phase input as they have in USA.  Even there, they don't mention anything about dropping phases.  It depends on your house wiring as to what you have available, so I would suggest only connecting one phase.  Your EV will still charge, but at 1/3 the rate of 3-phase.

You must have a very powerful solar inverter if it supports 3-phase.  Mine is only single phase, but with a 12kWh house battery, it keeps us running right through the night for much of the year.  In winter time, we charge our EV from the grid, but try to avoid it from September until April.

Well, I hope some of my comments will be useful to you.

Kind regards,
Grant.

**Reply by andre-richter** [reply]
> Can your EV actually handle 22kW of charging power though?

Currently I have a PHEV which supports single phase only. So even with the Pulsar Plus 22 kW I should be fine with this setup, because the car would only draw power from one phase, meaning minimum charging power starts at `1380 W`.

I am trying to future proof, though. I'd like to avoid a scenario where getting a BEV in the future that supports 3-phase also means minimum charging power raises to `4140 W` because of the way the Pulsar Plus works. But from the looks of it, this would be the case =/

**Reply by drc38** [reply]
Don't forget you can always set the charge current to 0A to give an effectively lower charge rate than 6A over say 5 or 10min (think slow version of PWM). 

My understanding of the reason for 6A is the charge efficiency (about 80% at 6A) and would drop rapidly below this as the car inverter consumes around 300W.

**Reply by GrantKeymer** [reply]
@andre-richter - Here's an update pointed out by @lbbrhzn in another thread...

> 
> Currently I have a PHEV which supports single phase only. So even with the Pulsar Plus 22 kW I should be fine with this setup, because the car would only draw power from one phase, meaning minimum charging power starts at `1380 W`.
> 
> I am trying to future proof, though. I'd like to avoid a scenario where getting a BEV in the future that supports 3-phase also means minimum charging power raises to `4140 W` because of the way the Pulsar Plus works. But from the looks of it, this would be the case =/

![Modified Eco Smart document](https://user-images.githubusercontent.com/35156198/147862332-e5aa8dc2-5813-4508-bfbd-dbfa833b71a6.png)

Notice the minimum of 460W per phase in Eco mode.  You need to install an EM530 energy meter to enable Eco mode but this looks like it could provide a solution for you :)

**Reply by andre-richter** [reply]
I've checked many other wallboxes in the meantime, and it indeed seems that 6A _per phase_ is as low as you can get for the minimum current, as you guys have indicated already. I've seen wallboxes that start only at 10A even.

I think I'll give Pulsar Plus with their own energy meter a try. It has the most important feature (charge only with 100% solar excess) implemented, and the _control path_ stays within your home. No stupid cloud service in between.

I initially thought about dynamically controlling the current over OCPP and Home Assistant, but if their energy meter does the job as well, I think it will relieve me from doing maintenance when my own implementation on Home Assistant breaks every few months 🙈 

Thanks a lot for your feedback @GrantKeymer @drc38 !

**Reply by GrantKeymer** [reply]
@andre-richter - I will be very interested to hear how you get on.  This is what happened when I tried to connect an EM112 to my Pulsar Plus.  The EM112 was purchased from Wallbox, so it should have been configured correctly.  Here is a link to the video I recorded...

https://drive.google.com/file/d/1y2jXiRCWxOZfXXwKmfMWknN6rpqmAzPJ/view?usp=sharing

Below is a screenshot showing which Upgrade Options are available - unfortunately NOT Power Boost or Eco-Smart.

![Wallbox App Upgrade Options Available](https://user-images.githubusercontent.com/35156198/147889507-d79e431c-413b-4c13-a7c5-0db4e20e39de.jpg)

With an EM112 connected, the Wallbox App should let me enable Power Boost and Eco-Smart as explained in the Installation Guide, but unfortunately it didn't work in my case.  As you say, it would be a very tidy solution if it worked.  Hopefully it will for you.

I am also worried about my implementation on HA breaking every few months, or maybe more often 🙈

So I will be very interested to hear what happens when you try Eco-Smart with a Pulsar Plus.

**Reply by lbbrhzn** [reply]
I can confirm both PowerBoost and Eco-Smart are working ok for my Pulsar Plus with EM340.

**Reply by andre-richter** [reply]
Thanks @lbbrhzn!

I was just thinking: If I get the 3-phase EM340 and the 22kW Pulsar, but connect my 1-phase EV.

Will the system understand only one phase is connected and start charging at 1380 W (full eco), or not understand only a single phase is connected and start charging only at 4140 W?

Maye I changed my mind again 🙄 and will start out with only Pulsar and try HA controlling.

**Reply by GrantKeymer** [reply]
@andre-richter - I think I can answer that one as I have a 3-phase Energy Meter similar to EM340 installed as part of our solar power system.  The energy meter adds power from all 3 phases and sends them out as one reading.  It doesn't measure each phase separately.

Therefore the Pulsar Plus will just read the total solar power available across all 3 phases and so long as there is enough, it will start charging at 1380W in Full Eco mode.  So long as your EV gets at least 6 Amps, it should be fine.

**Reply by andre-richter** [reply]
Ah, very interesting, thanks!

So in this case it seems the question is: What is the threshold value that the SW in a 3-phase Pulsar is looking for IF you connect a 1-phase vehicle — 1380 or 4140 kW?

Does it have built in intelligence to differentiate between 1-phase and 3-phase vehicle that gets connected 🤔

**Reply by GrantKeymer** [reply]
I think you will find it is like most 3-phase energy meters.  It doesn't care whether the output is across all 3 phases, or only one phase.  It adds the total power together according to the following statement from the Wallbox Academy article @lbbrhzn mentioned 2 days ago...

"**Eco-smart charging**

Eco-Smart checks the total active power in L1+L2, and starts charging with the following power: in Eco mode at 460W per phase and in Full Green at 1380W per phase."

I suspect this paragraph has been copied from the US version where they don't have 3-phase available in most homes.  Hence it only mentions L1+L2 whereas it should say L1+L2+L3.  We have 3-phase available at our property with a limit of 65 Amps per phase.  However, I haven't connected it to the Pulsar Plus for various reasons.

With a maximum 65 Amps available, I don't expect I'll ever need the Power Boost limiting function.  But Eco-Smart would have been handy, if it worked.  I still think HA will be a better option, allowing far more flexibility.

**Reply by andre-richter** [reply]
I might be doing a poor job expressing my issue. Let me try to paraphrase.

The key sentence in the official document is:
> Eco-Smart checks the total active power in L1+L2, and starts charging with the following power: in Eco mode at 460W per phase and in Full Green at 1380W per phase.

With my vehicle being 1-phase, there are two possible scenarios:

1. I get the 1-phase Pulsar (with 1-phase energy meter). EM sends available power to Pulsar; Pulsar checks for at least `1380W` (in full green) before starting to charge. Everything good.
2. I get the 3-phase Pulsar (with 3-phase energy meter). EM sends 3-phase aggregate of available power to Pulsar; Since the Pulsar is a 3-phase version, what does it do?
    1. It has built-in logic that recognizes there is only a 1-phase vehicle connected, and checks the EM report for at least `1380W`.
    2. It just flat out assumes 3-phase charging, since it is a 3-phase wallbox, and checks for `4140 W` as a threshold condition. Disregarding there is only a 1-phase vehicle connected. Meaning that charging the car in full green would start way too late.

If it is `2.ii`, this would be bad for my use case as long as I have the 1-phase car.

**Reply by lbbrhzn** [reply]
You can contact Wallbox support to find out how Eco-Smart works in detail for a 1-phase car. If is 2.ii, then you might be able to use  Eco mode instead of Full Green.

**Reply by GrantKeymer** [reply]
Sorry, I was getting mixed up in my thinking between 'Eco' mode and 'Full Green' mode.  I guess Wallbox have good reasons for their decision regarding minimum power levels, but I can see what you mean...

With a 3-Phase Pulsar, the Full Green minimum power threshold of 4140W is too demanding and unlikely to be met, except in summer time for a few hours per day.  The rest of the time, you would be stuck with Eco mode, pulling a fair amount of power from the grid and a smaller proportion of solar power.

To me, Wallbox's approach is too simplistic.  As I explained earlier, I want to have the flexibility of diverting house battery charging power to the EV when the sun is at its strongest.  Then, later in the afternoon, when the sun is diminishing, I want to stop charging the EV and divert all solar power to the house battery.  This is what I do now by using the Wallbox app.  Only with HA, will I get the flexibility I need and I expect your use case will be similar.

**Reply by lbbrhzn** [reply]
@GrantKeymer , which device do you intend use to decide what the charging current should be? In my setup, I use the P1 port of the smart meter that's installed by the electricity company.  I noticed that some people have connected a EM340 to their home assistant setup, using a RS485 interface board. See https://community.home-assistant.io/t/how-do-i-get-correct-values-from-carlo-gavazzi-em340-modbus/331909

**Reply by GrantKeymer** [reply]
> @GrantKeymer , which device do you intend use to decide what the charging current should be? In my setup, I use the P1 port of the smart meter that's installed by the electricity company.

@lbbrhzn - That's a good question!  First I'm going to try and get some details of the API from my solar power company.  If they don't have an API, I'll write a python script to scrape the values from the webpage below...

![Maxing out the Solar Array 8](https://user-images.githubusercontent.com/35156198/147925643-54493e29-33ca-4c63-8f2b-02b587843db9.jpg)

If neither of those approaches work out, I am thinking of trying a device from IAMMETER called WEM3080-150A.  It lets me get away from RS485 and use WiFi directly, also someone has developed an API suitable for HA.  However, it's interesting to see the info you mentioned about using an EM340 with HA.  Seems like there are plenty of options for Energy Meters with HA.  However, if I can use the existing energy meters (3 of them) provided with my solar power system, it makes sense to do so.

**Reply by andre-richter** [reply]
> I might be doing a poor job expressing my issue. Let me try to paraphrase.
> 
> The key sentence in the official document is:
> 
> > Eco-Smart checks the total active power in L1+L2, and starts charging with the following power: in Eco mode at 460W per phase and in Full Green at 1380W per phase.
> 
> With my vehicle being 1-phase, there are two possible scenarios:
> 
>     1. I get the 1-phase Pulsar (with 1-phase energy meter). EM sends available power to Pulsar; Pulsar checks for at least `1380W` (in full green) before starting to charge. Everything good.
> 
>     2. I get the 3-phase Pulsar (with 3-phase energy meter). EM sends 3-phase aggregate of available power to Pulsar; Since the Pulsar is a 3-phase version, what does it do?
>        
>        1. It has built-in logic that recognizes there is only a 1-phase vehicle connected, and checks the EM report for at least `1380W`.
>        2. It just flat out assumes 3-phase charging, since it is a 3-phase wallbox, and checks for `4140 W` as a threshold condition. Disregarding there is only a 1-phase vehicle connected. Meaning that charging the car in full green would start way too late.
> 
> 
> If it is `2.ii`, this would be bad for my use case as long as I have the 1-phase car.

In the meantime, I got feedback from Wallbox support. It currently really behaves like described in `2.ii`... 😞 

However, they said they are working on implementing 3 phas to 1 phase switching. But I cannot say if this will be a pure SW update for existing Wallboxes or needs new HW.

**Reply by GrantKeymer** [reply]
How about using some relays to drop out 1 or 2 phases?  You would need relays rated to handle 32 Amps and they aren't that expensive.  Then control them from a script in HA.

Good to hear you got a response from Wallbox in such a short time.  In my experience, they have been very slow.

**Comment by andre-richter** [comment]
> If is 2.ii, then you might be able to use Eco mode instead of Full Green.

Good point. Sounds like a possible workaround.
